### PR TITLE
Improve controllers test coverage

### PR DESCRIPTION
**Pull request by alaxalves**
Make sure these boxes are checked before your pull request is ready to be reviewed and merged. Thanks!

* [x] tests pass -- `rake test`
* [x] code is in uniquely-named feature branch, and has been rebased on top of latest master (especially if you've been asked to make additional changes)
* [x] pull request are descriptively named
* [ ] if possible, multiple commits squashed if they're smaller changes
* [ ] reviewed/confirmed/tested by another contributor or maintainer

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/wiki/contributing-to-public-lab-software

Thanks!

### CODE REVIEW

**Comment by alaxalves** [comment]
For some reason, the build has run on Travis, but it did not update on GH. Build link: https://travis-ci.org/github/publiclab/spectral-workbench/builds/693489215

File: app/controllers/comments_controller.rb [hunk 1]
```diff
@@ -27,12 +27,13 @@ def spectrum
...
         format.json  {
           @comment.body = RDiscount.new(@comment.body).to_html
+          flash[:notice] = "Comment saved."
           render :json => @comment
         }
       end
 
     else
-      flash[:error] == "There was an error creating your comment."
+      flash[:error] = "There was an error creating your comment."
```
Comment (jywarren) [inline app/controllers/comments_controller.rb:36]: omg how old are these bugs!!! 🤦 🎉

File: test/functional/sessions_controller_test.rb [hunk 2]
```diff
@@ -8,4 +8,88 @@ class SessionsControllerTest < ActionController::TestCase
 
   fixtures :users
 
+  test "should redirect if logged in" do
```
Comment (jywarren) [inline test/functional/sessions_controller_test.rb:11]: these are awesome

**Comment by jywarren** [comment]
So odd. i think we have to restart the build by closing and reopening...

**Comment by jywarren** [comment]
This is very strange. @alaxalves -- i'm sorry, but would you mind rebasing this one more time? I've seen this happen before and sometimes a clean rebase and force push can resolve it. Thank you!

**Pull request closed without merging**